### PR TITLE
Replace structopt with clap 3

### LINKED ISSUE #806: Update to use clap 3.0
As @didier-wenzek pointed out in https://github.com/thin-edge/thin-edge.io/discussions/801#discussioncomment-2088232 there's a new version of `clap` available that contains the `structopt` features, making `structopt` obsolete.

Thus, everything should be upgraded to `clap 3.0`.

---

I already started working on this, but cannot yet publish my changes because of the outstanding issue we discussed regarding the CLA.
I opened this issue merely for documentation that I started work on this, so we do not collide in our efforts :laughing:

### PR DESCRIPTION
**Pull request by matthiasbeyer**
## Proposed changes

Closes #806 

This is implemented in one patch, as I see very little benefit in splitting this per crate. The changes are absolutely minimal, only replacing each instance of `structopt` with `clap`. No other code is touched.

## Types of changes

- [x] Improvement

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
It seems that [this failure](https://github.com/thin-edge/thin-edge.io/runs/5228391353?check_suite_focus=true#step:7:444) is due to the fact that the dummy plugin does not have a `update-list` subcommand.

Should I add one? It seems that the dummy plugin does the same thing regardless what command was issued, so the change would be trivial (read: I can do it on my own :laughing: )

**Review by makr11st — APPROVED** [review]

**Comment by makr11st** [comment]
> Should I add one? It seems that the dummy plugin does the same thing regardless what command was issued, so the change would be trivial (read: I can do it on my own 😆 )

Please do add one to align the plugin and remove the error.

File: crates/core/tedge/src/cli/connect/cli.rs [hunk 1]
```diff
@@ -33,7 +32,7 @@ impl BuildCommand for TEdgeConnectOpt {
                 cloud: Cloud::C8y,
                 common_mosquitto_config: CommonMosquittoConfig::default(),
                 is_test_connection,
-                service_manager: service_manager(
+                service_manager: crate::system_services::service_manager(
```
Comment (didier-wenzek) [inline crates/core/tedge/src/cli/connect/cli.rs:35]: I would be good to remove the now useless `use crate::system_services::service_manager;`

**Review by didier-wenzek — COMMENTED** [review]
Just some minor comments.

* `main.rs` is using the deprecated function  `clap::Parser::from_args`

File: crates/core/tedge/src/cli/connect/cli.rs (hunk 1 above)
Comment (matthiasbeyer) [reply crates/core/tedge/src/cli/connect/cli.rs:35]: I guess the other way round would be better, not using the full path here! :+1:

File: crates/core/tedge_mapper/Cargo.toml [hunk 2]
```diff
@@ -43,7 +43,7 @@ mqtt_channel = { path = "../../common/mqtt_channel" }
 reqwest = { version = "0.11", default-features = false, features = ["json", "rustls-tls"] }
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
-structopt = "0.3"
+clap = { version = "3.0", features = ["cargo", "derive"] }
```
[the remark this answers is not in the document]
Comment (matthiasbeyer) [reply crates/core/tedge_mapper/Cargo.toml:46]: Thanks for noticing, this slipped through.

**Comment by matthiasbeyer** [comment]
Pushed fixups, will rebase before merge, of course!

**Review by didier-wenzek — APPROVED** [review]
Approved.

The flappy test `get_jwt_token_full_run` is really annoying.
* => I relaunch the check for this PR.
* => If no fix is found today I will remove this test.

**Comment by matthiasbeyer** [comment]
I resolved the merge conflict and squashed my fixups. Ready to merge from my side :+1: !

**Comment by matthiasbeyer** [comment]
I assume that you're okay with me merging this. If not, we can always revert the merge again... but since you approved I guess this is alright! :+1:

**Pull request merged**